contracts/Blast_SpokePool.sol

```diff
@@ -0,0 +1,118 @@
+
+// USDB and WETH on Blast accrue yield that can be claimed by any account holding the token. So for the length of
+// time that the SpokePool holds on to these assets, it can can claim interest.
+interface IERC20Rebasing {
+    function claim(address recipient, uint256 amount) external returns (uint256);
+
+    function getClaimableAmount(address account) external view returns (uint256);
+}
+
+/**
+ * @notice Blast Spoke pool.
+ */
+contract Blast_SpokePool is Ovm_SpokePool {
+    // This is the yield-accruing stablecoin on Blast that USDC/DAI/USDT all bridge into. It can be withdrawn
+    // from L2 into DAI.
+    address private constant USDB = 0x4300000000000000000000000000000000000003;
+    // Token that is received when withdrawing USDB, aka DAI.
+    address private constant L1_USDB = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address private constant L2_BLAST_BRIDGE = 0x4300000000000000000000000000000000000005;
+
+    error InvalidClaimedAmount(address token);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address _wrappedNativeTokenAddress,
+        uint32 _depositQuoteTimeBuffer,
+        uint32 _fillDeadlineBuffer,
+        IERC20 _l2Usdc,
+        ITokenMessenger _cctpTokenMessenger
+    )
+        Ovm_SpokePool(
+            _wrappedNativeTokenAddress,
+            _depositQuoteTimeBuffer,
+            _fillDeadlineBuffer,
+            _l2Usdc,
+            _cctpTokenMessenger
+        )
+    {} // solhint-disable-line no-empty-blocks
+
+    /**
+     * @notice Construct the OVM Blast SpokePool.
+     * @param _initialDepositId Starting deposit ID. Set to 0 unless this is a re-deployment in order to mitigate
+     * relay hash collisions.
+     * @param _crossDomainAdmin Cross domain admin to set. Can be changed by admin.
+     * @param _hubPool Hub pool address to set. Can be changed by admin.
+     */
+    function initialize(
+        uint32 _initialDepositId,
+        address _crossDomainAdmin,
+        address _hubPool
+    ) public initializer {
+        __OvmSpokePool_init(_initialDepositId, _crossDomainAdmin, _hubPool, Lib_PredeployAddresses.OVM_ETH);
+    }
+
+    /**
+     * @notice Claim interest for token into this contract. This should be called before _bridgeTokensToHubPool
+     * and then the claimed amount should be added to the bridged amount.
+     */
+    function _claimYield(IERC20Rebasing token) internal returns (uint256 claimedAmount) {
+        uint256 claimableAmount = token.getClaimableAmount(address(this));
+        claimedAmount = token.claim(address(this), claimableAmount);
+        if (claimableAmount != claimedAmount) {
+            revert InvalidClaimedAmount(address(token));
+        }
+    }
+
+    /**
+     * @notice Claims any yield for tokens that accrue yield and then also bridges those
+     */
+    function _bridgeTokensToHubPool(uint256 amountToReturn, address l2TokenAddress) internal override {
+        uint256 amountToWithdraw = amountToReturn;
```

Comment:
nit: why define a new variable? Why not just increment the value passed in? Note: integers are passed by value (copied), so this won't and can't impact any variables in the caller's context